### PR TITLE
fixed bluespace harvester not reconnecting to powernet properly

### PR DESCRIPTION
**Pull request by chc**
## What Does This PR Do

This PR fixes issue #23372 where the bluespace harvester cannot be attached to a new powernet. While this issue is not yet confirmed, it is easy enough to verify by simply attaching a harvester to a powernet (for example the turbine SMES charge powernet), and then join the turbine SMES charge powernet to the station powernet. The harvester will never draw load again despite there being sufficient power, or even if you reattach it to the old powernet.

## Why It's Good For The Game

It is very frustrating to build a harvester, only for it to be unusable.

## Testing

Reproduced steps mentioned above about joining to turbine power net on master branch, and observed harvester no longer drawing load, and then repeated again after these changes to see it working properly in all caes.

## Changelog
:cl:
fix: Fixed bluespace harvester not reconnecting to powernet properly
/:cl:

### CODE REVIEW

File: code/modules/station_goals/bluespace_tap.dm [hunk 1]
```diff
@@ -316,8 +317,9 @@
 
 
 /obj/machinery/power/bluespace_tap/connect_to_network()
-	..()
+	var/to_return = ..()
 	update_icon()
+	return to_return
```
Comment (Sirryan2002) [inline code/modules/station_goals/bluespace_tap.dm:322]:
```suggestion
	. = ..()
	if(.)
		update_icon()
```
BYOND has a native return variable called `.` which will automatically return at the end of the proc.

**Review by Sirryan2002 — APPROVED** [review]
good PR

File: code/modules/station_goals/bluespace_tap.dm (hunk 1 above)
Comment (DGamerL) [inline code/modules/station_goals/bluespace_tap.dm:322]:
```suggestion
	. = ..()
	update_icon()
```
`.` is a variable that's already automatically returned!

File: code/modules/station_goals/bluespace_tap.dm [hunk 2]
```diff
@@ -197,6 +197,7 @@
 	density = TRUE
 	interact_offline = TRUE
 	luminosity = 1
+	anchored = TRUE
```
Comment (DGamerL) [inline code/modules/station_goals/bluespace_tap.dm:200]: `machinery/power` is anchored by default, does this do anything or is the fix solely the `connect_to_network()`?

**Review by DGamerL — CHANGES_REQUESTED** [review]

File: code/modules/station_goals/bluespace_tap.dm (hunk 2 above)
Comment (chc) [reply code/modules/station_goals/bluespace_tap.dm]: thanks thats removed

File: code/modules/station_goals/bluespace_tap.dm (hunk 1 above)
Comment (chc) [reply code/modules/station_goals/bluespace_tap.dm]: i added that change and verified the icon updates as expected

**Review by DGamerL — APPROVED** [review]

**Pull request closed without merging**